src/app-manager.ts

```diff
@@ -130,12 +134,12 @@ export class AppManager {
 
     const methods = json.methods || []
     methods.forEach(key => {
-      if (key in api) delete this[api.type][key]
+      if (key in api) delete this[api.type as string][key]

```

Comment:
why do we need `as string` here? `type` is not already enforced to be a string as it is declared in the interface `API`?

Reply:
In this case, it's kind of a mismatch between the generic Api "M" in this method and what the `this.calls` method accept with the generic IAppManager "T" in the class. Removing the `as string` create a typescript warning. It's a quite fix to remove a Typescript warning, it won't impact the end Javascript code. But yes, I should improve that in time.